halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/SpinnakerMonitoringProfile.java

```diff
@@ -0,0 +1,140 @@
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerEndpoints;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.SpinnakerMonitoringService;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.SpinnakerService;
+import io.fabric8.utils.Strings;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@Component
+public class SpinnakerMonitoringProfile extends SpinnakerProfile {
+  private String profileFileName = "spinnaker-monitoring.yml";
+
+  @Override
+  public String commentPrefix() {
+    return "## ";
+  }
+
+  @Override
+  public SpinnakerArtifact getArtifact() {
+    return SpinnakerArtifact.SPINNAKER_MONITORING;
+  }
+
+  @Override
+  protected ProfileConfig generateFullConfig(ProfileConfig config, DeploymentConfiguration deploymentConfiguration, SpinnakerEndpoints endpoints) {
+    String primaryConfig = config.getPrimaryConfigFile();
+    SpinnakerEndpoints.Services services = endpoints.getServices();
+    SpinnakerMonitoringService monitoringService = services.getSpinnakerMonitoring();
+    MetricStores metricStores = deploymentConfiguration.getMetricStores();
+    List<String> enabledMetricStores = new ArrayList<>();
+
+    DatadogStore datadogStore = metricStores.getDatadog();
+    if (datadogStore.isEnabled()) {
+      enabledMetricStores.add("datadog");
+    }
+
+    PrometheusStore prometheusStore = metricStores.getPrometheus();
+    if (prometheusStore.isEnabled()) {
+      enabledMetricStores.add("prometheus");
+    }
+
+    StackdriverStore stackdriverStore = metricStores.getStackdriver();
+    if (stackdriverStore.isEnabled()) {
+      enabledMetricStores.add("stackdriver");
+    }
+
+    config.extendConfig(primaryConfig, yamlToString(metricStores));
+
+    Server server = new Server()
+        .setHost(monitoringService.getHost())
+        .setPort(monitoringService.getPort());
+
+    ServerConfig serverConfig = new ServerConfig();
+    serverConfig.setServer(server);
+
+    config.extendConfig(primaryConfig, yamlToString(serverConfig));
+
+    Monitor monitor = new Monitor()
+        .setPeriod(metricStores.getPeriod())
+        .setMetricStore(enabledMetricStores);
+
+    MonitorConfig monitorConfig = new MonitorConfig();
+    monitorConfig.setMonitor(monitor);
+
+    config.extendConfig(primaryConfig, yamlToString(monitorConfig));
+
+    for (SpinnakerService spinnakerService : services.allServices()) {
+      if (!spinnakerService.isMonitoringEnabled()) {
+        continue;
+      }
+
+      // The tacit assumption is that the monitoring daemon will always be installed
+      // on whatever machine the service it's monitoring is running.
+      String metricsConfig = "metrics_url: http://localhost:" + spinnakerService.getPort() + "/spectator/metrics";
```

Comment:
This will need to be more forgiving - specifically with Gate, where you'll need to add HTTP Basic Auth when trying to extract metrics.